src/Features/CSharp/Portable/CodeRefactorings/ConvertIfToSwitch/CSharpConvertIfToSwitchCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,246 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Composition;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CodeRefactorings.ConvertIfToSwitch;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.LanguageServices;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeRefactorings.ConvertIfToSwitch
+{
+    [ExportCodeRefactoringProvider(LanguageNames.CSharp, Name = nameof(CSharpConvertIfToSwitchCodeRefactoringProvider)), Shared]
+    internal sealed partial class CSharpConvertIfToSwitchCodeRefactoringProvider : AbstractConvertIfToSwitchCodeRefactoringProvider
+    {
+        protected override IAnalyzer CreateAnalyzer(ISyntaxFactsService syntaxFacts, SemanticModel semanticModel)
+            => new CSharpAnalyzer(syntaxFacts, semanticModel);
+
+        private sealed class CSharpAnalyzer : Analyzer<StatementSyntax, IfStatementSyntax, ExpressionSyntax>
+        {
+            private int _numberOfSubsequentIfStatementsToRemove = 0;
+
+            public CSharpAnalyzer(ISyntaxFactsService syntaxFacts, SemanticModel semanticModel)
+                : base(syntaxFacts, semanticModel)
+            {
+            }
+
+            protected override string Title => CSharpFeaturesResources.Convert_if_to_switch;
+
+            protected override IPattern CreatePatternFromExpression(ExpressionSyntax operand)
+            {
+                switch (operand.Kind())
+                {
+                    case SyntaxKind.EqualsExpression:
+                        {
+                            // Look for the form "x == 5" or "5 == x" where "x" is equivalent to the switch expression.
+                            // This will turn into a constant pattern e.g. "case 5:".
+                            var node = (BinaryExpressionSyntax)operand;
+                            if (!TryDetermineConstant(node.Right, node.Left,
```

Comment:
hrmm... maybe it doesn't... since you verify that the expression must have constant value..